### PR TITLE
Re Input Augmentation: Gaussian noise on log(Re) for OOD robustness

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The model receives Reynolds number as a log(Re) input channel. During training, the model sees a fixed distribution of Re values from the training set. At test time, p_re evaluates on OOD Reynolds numbers — the model must extrapolate its learned Re-pressure relationship to unseen values.

By adding Gaussian noise to the log(Re) input channel during training, we force the model to be **less sensitive to exact Re values** and instead learn pressure patterns that are robust across a range of Re. This is the standard "input noise regularization" technique adapted specifically for the Re channel.

**Physical motivation:** For most mesh nodes (volume nodes far from walls, mid-chord surface nodes), the pressure field changes slowly with Re. Only nodes in the boundary layer and separation region are highly Re-sensitive. Adding Re noise teaches the model to focus on geometry-driven pressure patterns rather than over-fitting to precise Re conditioning — a form of OOD robustness training.

**Analogy:** `--aug aoa_perturb` already adds noise to the AoA input, which has proven effective (part of the established baseline). This extends the same principle to Reynolds number.

**Key difference from failed Re-scaled walldist (#2228):** That PR added a redundant feature (product of existing inputs). This PR adds input NOISE during training — a regularization technique, not an input feature. No new parameters, no new channels. Just a training-time perturbation that widens the effective training distribution.

**Expected improvement:** -2 to -4% p_re (OOD Reynolds). Possible secondary benefit on p_oodc (OOD condition, which may include Re variation).

**Risk:** If the model needs precise Re information for in-distribution accuracy, adding noise could hurt p_in. We mitigate by using small sigma (0.1 in log space ≈ ±10% Re) and applying noise ONLY during training (not validation).

**Confidence:** Medium. Standard ML technique (input noise regularization), targeted at the weakest metric. Main uncertainty is optimal noise magnitude.

## Instructions

All changes in `cfd_tandemfoil/train.py`.

### Step 1: Add config flags

```python
aug_re_noise: bool = False     # Gaussian noise on log(Re) input during training
aug_re_sigma: float = 0.1     # std of noise in log(Re) space (~±10% Re)
```

### Step 2: Identify where log(Re) enters the input

Find where the Reynolds number is encoded into the input features. It's likely one of the input channels in `x[:, :, k]`. Check how `log_Re` or `re` is computed and added to the feature vector. Note the exact channel index.

### Step 3: Add noise during training

In the **training loop only** (NOT in val/vis/verify), after the input features are assembled but BEFORE normalization:

```python
if cfg.aug_re_noise and model.training:
    # Find the Re channel index — adjust based on your investigation in Step 2
    re_channel_idx = <discovered_index>
    
    # Add Gaussian noise to log(Re) in the training input
    re_noise = torch.randn(x.shape[0], 1, 1, device=x.device) * cfg.aug_re_sigma
    # Broadcast: same noise for all nodes in a sample (Re is global)
    x[:, :, re_channel_idx] = x[:, :, re_channel_idx] + re_noise.squeeze(-1)
```

**IMPORTANT:** The noise must be:
- Per-sample (same noise for all nodes in a batch element)
- Only during training (disabled during validation/visualization)
- Applied to the log(Re) channel specifically (not other inputs)

### Step 4: Run 2 seeds

```bash
# Seed 42
cd cfd_tandemfoil && python train.py \
  --agent edward --wandb_name "edward/re-aug-s42" \
  --wandb_group "round18/re-input-augmentation" \
  --seed 42 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature \
  --aug_re_noise --aug_re_sigma 0.1

# Seed 73 — identical but --seed 73 --wandb_name "edward/re-aug-s73"
```

### Step 5: Report results

Table: p_in, p_oodc, p_tan, p_re for both seeds, 2-seed avg, baseline comparison, W&B run IDs. **Pay special attention to p_re** — that's the target metric. Also note if p_in degrades (would indicate noise is too large).

## Baseline

Current best (PR #2213, Wake Deficit Feature, 2-seed average):

| Metric | Baseline | Target to beat |
|--------|----------|----------------|
| p_in   | **11.979** | < 11.98 |
| p_oodc | **7.643**  | < 7.65  |
| **p_tan** | **28.341** | **< 28.34** |
| p_re   | **6.300**  | < 6.30  |

W&B runs: `hgml7i2r` (seed 42), `qic03vrg` (seed 73)

**Reproduce baseline:**
```bash
cd cfd_tandemfoil && python train.py --agent edward --wandb_name "edward/baseline-wake-deficit" \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT edward:

## Results

**Hypothesis**: Adding Gaussian noise (σ=0.1) to the log(Re) input channel during training to improve OOD Reynolds number generalization.

**Implementation**: 3-line change — per-sample Gaussian noise on channel 13 (log_Re) in training loop only, before normalization. No new parameters, no new channels.

### Metrics

| Metric | Seed 42 | Seed 73 | 2-seed avg | Baseline | Delta |
|--------|---------|---------|-----------|----------|-------|
| surf p_in   | 12.180 | 12.490 | 12.335 | 11.979 | +3.0% ❌ |
| surf p_tan  | 28.300 | 28.165 | 28.232 | 28.341 | **-0.4%** ✅ |
| surf p_oodc |  8.031 |  7.814 |  7.922 |  7.643 | +3.7% ❌ |
| surf p_re   |  6.548 |  6.620 |  6.584 |  6.300 | +4.5% ❌ |

**W&B run IDs**: `vb5xooea` (s42, edward/re-aug-s42), `ix79j0bv` (s73, edward/re-aug-s73)
**W&B group**: `round18/re-input-augmentation`

**Command**:
```bash
cd cfd_tandemfoil && python train.py --agent edward --wandb_name "edward/re-aug-s42" \
  --wandb_group "round18/re-input-augmentation" --seed 42 \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --pcgrad_3way --pcgrad_extreme_pct 0.15 \
  --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3 \
  --aft_foil_srf --aug_gap_stagger_sigma 0.02 --aug_dsdf2_sigma 0.05 \
  --gap_stagger_spatial_bias \
  --dct_freq_loss --dct_freq_weight 0.05 --dct_freq_gamma 2.0 --dct_freq_alpha 1.5 \
  --te_coord_frame --wake_deficit_feature \
  --aug_re_noise --aug_re_sigma 0.1
# Seed 73: identical with --seed 73 --wandb_name "edward/re-aug-s73"
```

**Epochs**: ~148 (s42), ~146 (s73) at ~68-75s/epoch (180-min timeout)
**Note**: Runs executed sequentially (single GPU) — s73 first, then s42 — due to OOM when running in parallel.

### What happened

**MIXED/NEGATIVE result — p_tan improved slightly but all other metrics degraded, including the target metric p_re.**

The Re noise augmentation (σ=0.1 in log-space, ~±10% Re) produced:
- **p_tan -0.4%**: Small improvement on tandem transfer, the only positive signal. Noise may help the model generalize tandem pressure distributions across Re.
- **p_re +4.5%**: The target metric degraded significantly — the opposite of what was expected. Adding noise to Re didn't improve OOD Re generalization; it hurt it.
- **p_in +3.0%, p_oodc +3.7%**: In-distribution and OOD-condition also degraded.

**Why it failed:**

1. **Re information is critical, not redundant**: Unlike AoA (where small perturbations teach geometric robustness), log(Re) is a highly informative conditioning signal that the model needs precise access to. σ=0.1 in log-space corresponds to Re varying by ±10%, which is substantial — the model may lose the ability to learn Re-dependent features accurately.

2. **Already-noisy signal**: The existing `--aug aoa_perturb` adds rotation noise to positions and DSDF, and `--aug_gap_stagger_sigma 0.02` adds gap/stagger noise. The model already has noise-based regularization in multiple input channels. Adding Re noise on top may push past the sweet spot.

3. **σ=0.1 may be too large**: The dataset log(Re) std is ~0.76 (from split_stats.json), so σ=0.1 is ~13% of the natural variation. This might be too much — a smaller σ (e.g., 0.02-0.05) could help without destroying signal.

4. **OOD Re is already strong**: p_re=6.300 baseline is already very good relative to the other metrics. The model's Re handling may not be the bottleneck.

### Suggested follow-ups

- **Smaller σ (0.02-0.05)**: Try much smaller noise magnitude to see if there's a sweet spot where regularization helps without destroying signal.
- **Curriculum noise**: Start with no noise and gradually increase σ during training — let the model learn precise Re features first, then regularize.
- **Noise only for volume nodes**: Surface nodes need precise Re for BL prediction. Apply noise only to volume nodes where Re sensitivity is lower.
- **Dropout-style Re masking**: Instead of adding noise, randomly zero-out the Re channel with some probability — teaches the model to predict from geometry alone sometimes.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2233 because ADVISOR: Closing — target metric p_re degraded +4.5% (opposite of hypothesis). p_tan saw a marginal -0.4% improvement but within noise. 3 of 4 metrics regressed. Your analysis about Re being critical (not redundant) is insightful — log(Re) is a highly informative conditioning signal that the model needs precise access to. σ=0.1 (~10% Re variation) is too aggressive. The model already has well-tuned targeted augmentations (AoA, DSDF, gap/stagger). Adding Re noise on top pushed past the regularization sweet spot. The baseline's OOD Re performance (6.300) may simply not be bottlenecked by Re sensitivity..
